client/src/components/job-lead-dashboard-component/job-lead-dashboard-filters/index.jsx

```diff
@@ -0,0 +1,463 @@
+              onChange={setStartDateExpired}
+              renderInput={(params) => {
+                // eslint-disable-next-line react/jsx-props-no-spreading
+                return <TextField {...params} size="small" fullWidth />;
+              }}
+            />
+            <DatePicker
+              label="Until"
+              value={endDateExpired}
+              onChange={setEndDateExpired}
+              renderInput={(params) => {
+                // eslint-disable-next-line react/jsx-props-no-spreading
+                return <TextField {...params} size="small" fullWidth />;
+              }}
+            />
+            {/* Date Expired Filter */}
+            <Typography
+              sx={{ fontSize: 14 }}
+              color="text.secondary"
+              align="left"
+              gutterBottom
+            >
+              Expiry Date
+            </Typography>
+            <DatePicker
+              label="From"
+              value={startDateCreated}
+              onChange={setStartDateCreated}
+              renderInput={(params) => {
+                // eslint-disable-next-line react/jsx-props-no-spreading
+                return <TextField {...params} size="small" fullWidth />;
+              }}
+            />
+            <DatePicker
+              label="Until"
+              value={endDateCreated}
+              onChange={setEndDateCreated}
+              renderInput={(params) => {
+                // eslint-disable-next-line react/jsx-props-no-spreading
+                return <TextField {...params} size="small" fullWidth />;
+              }}
+            />
+            {/* Compensation Filter */}
+            <Typography
+              sx={{ fontSize: 14 }}
+              color="text.secondary"
+              align="left"
+              gutterBottom
+            >
+              Compensation
+            </Typography>
+            <Box sx={{ px: 2 }}>
+              <Slider
+                value={compensationRange}
+                onChange={onCompensationRangeChange}
+                valueLabelDisplay="auto"
+                getAriaValueText={valuetext}
+                disableSwap
+                step={1}
+                min={minCompensation}
+                max={maxCompensation}
+                marks={compensationSliderMarks}
+              />
+            </Box>
+
+            {/* Hours Per Week Filter */}
+            <Typography
+              sx={{ fontSize: 14 }}
+              color="text.secondary"
+              align="left"
+              gutterBottom
+            >
+              Hours/week
+            </Typography>
+            <Box sx={{ px: 2 }}>
+              <Slider
+                value={hoursPerWeekRange}
+                onChange={onHoursPerWeekRangeChange}
+                valueLabelDisplay="auto"
+                getAriaValueText={valuetext}
+                disableSwap
+                step={1}
+                min={minHoursPerWeek}
+                max={maxHoursPerWeek}
```

Comment:
Both min and max should be set 0 and 40. This enables us to always be in range when searching. You might need to update the values you set it to to work with this functionality but it should solve a lot of the issues we have been getting

Reply:
Hey Jordan, I figured out the issue why the boundary was weird, I have implemented a fix that shouldn't have this be an issue anymore. Essentially, from the backend, we pass to the frontend the min and max values from the entire table and then we use those values to set the boundary instead of calculating it in the frontend.